element_array_ephys/ephys_organoids.py

```diff
@@ -128,382 +129,185 @@ class AcquisitionSoftware(dj.Lookup):
-                )
 
-            if probe_data.probe_model in supported_probe_types:
-                probe_type = probe_data.probe_model
-                electrode_query = probe.ProbeType.Electrode & {"probe_type": probe_type}
+@schema
+class EphysSessionInfo(dj.Imported):
+    definition = """
+    -> EphysSession
+    attribute_name      : varchar(32)
+    ---
+    attribute_blob=null : longblob
+    """
 
-                probe_electrodes = {
-                    key["electrode"]: key for key in electrode_query.fetch("KEY")
-                }
+    @property
+    def key_source(self):
+        return EphysSession - "session_type='spike_sorting'"
 
-                electrode_group_members = [
-                    probe_electrodes[channel_idx]
-                    for channel_idx in probe_data.ap_meta["channels_indices"]
-                ]
-            else:
-                raise NotImplementedError(
-                    "Processing for neuropixels"
-                    " probe model {} not yet implemented".format(probe_data.probe_model)
-                )
-
-            self.insert1(
+    def make(self, key):
+        file = (RawData & key).fetch("file_path", order_by="start_time")[0]
+        data = intanrhsreader.load_file(file)
+        del data["header"], data["t"]
+        self.insert(
+            [
                 {
                     **key,
-                    **generate_electrode_config(probe_type, electrode_group_members),
-                    "acq_software": acq_software,
-                    "sampling_rate": probe_data.ap_meta["sample_rate"],
-                    "recording_datetime": probe_data.recording_info[
-                        "recording_datetimes"
-                    ][0],
-                    "recording_duration": np.sum(
-                        probe_data.recording_info["recording_durations"]
-                    ),
+                    "attribute_name": k,
+                    "attribute_blob": v,
                 }
-            )
-
-            root_dir = find_root_directory(
-                get_ephys_root_data_dir(),
-                probe_data.recording_info["recording_files"][0],
-            )
-            self.EphysFile.insert(
-                [
-                    {**key, "file_path": fp.relative_to(root_dir).as_posix()}
-                    for fp in probe_data.recording_info["recording_files"]
-                ]
-            )
-            # explicitly garbage collect "dataset"
-            # as these may have large memory footprint and may not be cleared fast enough
-            del probe_data, dataset
-            gc.collect()
-        else:
-            raise NotImplementedError(
-                f"Processing ephys files from"
-                f" acquisition software of type {acq_software} is"
-                f" not yet implemented"
-            )
+                for k, v in data.items()
+                if "data" not in k
```

Comment:
Since you are deleting the `header` and `t` keys above, should we also just delete the `data` key?

Reply:
Essentially this is to remove any keys that contain `data` string, not just `data`. They tend to be quite long so I thought they shouldn't be stored here.

Reply:
Thanks for clarifying.  That makes sense.  We can ignore my comment.